### PR TITLE
Stops putting RCD, bluespace bags, and staves into ORM

### LINKED ISSUE #30764: The ORM will consume an rcd if you click with it
You insert a material total of 100000 into the ore redemption machine.

This deletes the RCD and as far as I can tell gives no resources

### PR DESCRIPTION
**Pull request by ShizCalev**
Closes #30764

### CODE REVIEW

File: code/datums/components/material_container.dm [hunk 1]
```diff
@@ -59,6 +59,9 @@
 	var/datum/callback/pc = precondition
 	if(pc && !pc.Invoke())
 		return
+	if(I.flags_2 & NO_MAT_REDEMPTION_2)
+		to_chat(user, "<span class='warning'>[parent] won't accept \the [I]!</span>")
```
Comment (Cyberboss) [inline code/datums/components/material_container.dm]: remove `\the`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/material_container.dm [hunk 2]
```diff
@@ -59,6 +59,9 @@
 	var/datum/callback/pc = precondition
```
Comment (Cyberboss) [inline code/datums/components/material_container.dm]: Actually, wait, there's a massive if up on L55 that runs before . is set to TRUE. Remove the message and just change the bit that says `I.flags_2 & HOLOGRAM_2` to `I.flags_2 & (HOLOGRAM_2 | NO_MAT_REDEMPTION_2)`

**Review by Cyberboss — CHANGES_REQUESTED** [review]

File: code/datums/components/material_container.dm (hunk 2 above)
Comment (ShizCalev) [reply code/datums/components/material_container.dm]: Oh hell, how'd I miss that.

**Comment by kevinz000** [comment]
Uh shouldn't you do this but not block it from autolathes? Recycling staves make no sense but rcd does.

**Review by Cyberboss — DISMISSED** [review]

File: code/datums/components/material_container.dm [hunk 3]
```diff
@@ -52,7 +52,7 @@
 
 /datum/component/material_container/proc/OnAttackBy(obj/item/I, mob/living/user)
 	var/list/tc = allowed_typecache
-	if(user.a_intent == INTENT_HARM || (I.flags_2 & HOLOGRAM_2) || (tc && !is_type_in_typecache(I, tc)))
+	if(user.a_intent == INTENT_HARM || (I.flags_2 & HOLOGRAM_2 | NO_MAT_REDEMPTION_2) || (tc && !is_type_in_typecache(I, tc)))
 		return FALSE
```
Comment (ShizCalev) [inline code/datums/components/material_container.dm]: @Cyberboss Actually, would it be cool if I added the chat message under here so it's not just a silent thing and people realize why nothing's happening when they click the machine?
```
if(!user.a_intent == INTENT_HARM)
    to_chat(user, "<span class='warning'>[parent] won't accept [I]!</span>")
```

Comment (Cyberboss) [reply code/datums/components/material_container.dm]: Not for the harm intent thing (that's supposed to hit X with Y). But, for the hologram and restrictions, yeah go ahead.

**Review by Cyberboss — APPROVED** [review]

**Pull request merged**